Solution1/ConsoleApp/Extension/RegisterDependencies.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Net.Http;
+using AutoMapper;
+using BusinessLayer.Abstract;
+using BusinessLayer.Service;
+using BusinessLayer.Service.Abstract;
+using ConsoleApp.AutoMap;
+using Ninject;
+
+namespace ConsoleApp.Extension
+{
+    public static class RegisterDependencies
+    {
+        public static void AddServices(this IKernel ninjectKernel)
+        {
+            ninjectKernel.Bind<IWeatherServiсe>().To<WeatherService>();
+            ninjectKernel.Bind<IWeatherApiService>().To<WeatherApiService>()
+                .WithConstructorArgument("httpClient", new HttpClient());
```

Comment:
use IHttpClientFactory for injecting dependency

Reply:
-